From 32d610fb70ec33495c1bb524de83cb7f7fd83b50 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 14 Dec 2018 13:25:11 +0100
Subject: [PATCH 39/50] DreamBackend: Fix launching settings

* Calling startActivity() from outside of an Activity
  context requires the FLAG_ACTIVITY_NEW_TASK flag.

Change-Id: I2f9499953d131c79487bc8b1fc76e386cdd0f3be
---
 .../src/com/android/settingslib/dream/DreamBackend.java      | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/dream/DreamBackend.java b/packages/SettingsLib/src/com/android/settingslib/dream/DreamBackend.java
index e5d97c9b4a7..8430526d8a6 100644
--- a/packages/SettingsLib/src/com/android/settingslib/dream/DreamBackend.java
+++ b/packages/SettingsLib/src/com/android/settingslib/dream/DreamBackend.java
@@ -260,7 +260,10 @@ public class DreamBackend {
         logd("launchSettings(%s)", dreamInfo);
         if (dreamInfo == null || dreamInfo.settingsComponentName == null)
             return;
-        mContext.startActivity(new Intent().setComponent(dreamInfo.settingsComponentName));
+        Intent intent = new Intent()
+                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                .setComponent(dreamInfo.settingsComponentName);
+        mContext.startActivity(intent);
     }
 
     public void preview(DreamInfo dreamInfo) {
-- 
2.17.1

